src/query/storages/table-meta/src/meta/common.rs

```diff
@@ -89,3 +91,62 @@ impl Compression {
         Compression::Lz4
     }
 }
+
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone, PartialEq, Eq, Default)]
+pub struct ColumnNDVs {
+    pub column_hll: HashMap<ColumnId, HyperLogLog<DataValue>>,
+}
+
+impl ColumnNDVs {
+    // The error rate of hll, the less error rate, the more hll size
+    // Error rate  hll size(bytes)
+    // 0.04         1057
+    // 0.09         289
+    // 0.11         161
+    const HLL_ERROR_RATE: f64 = 0.11;
+
+    pub fn merge_number_of_distinct_values(&mut self, other: &ColumnNDVs) {
+        for (idx, other_hll) in other.column_hll.iter() {
+            match self.column_hll.get_mut(idx) {
+                Some(hll) => hll.union(other_hll),
+                None => {
+                    self.column_hll.insert(*idx, other_hll.clone());
+                }
+            }
+        }
+    }
+
+    pub fn add_column(&mut self, idx: u32, col: &Arc<dyn Column>) {
```

Comment:
Do you have any performance numbers about making all column with HLL indexes ?

I suggest to load large datasets into databend and have the comparisions.

https://databend.rs/doc/learn/analyze-hits-dataset-with-databend